src/cli/java/org/commcare/util/screen/MultiSelectEntityScreen.java

```diff
@@ -0,0 +1,84 @@
+package org.commcare.util.screen;
+
+import static org.commcare.session.SessionFrame.STATE_MULTIPLE_DATUM_VAL;
+
+import org.commcare.core.interfaces.EntitiesSelectionCache;
+import org.commcare.modern.session.SessionWrapper;
+import org.commcare.session.CommCareSession;
+import org.commcare.suite.model.MultiSelectEntityDatum;
+import org.javarosa.core.model.instance.TreeReference;
+import org.javarosa.core.util.PropertyUtils;
+
+import java.sql.SQLException;
+
+import javax.annotation.Nullable;
+
+/**
+ * Variation of EntityScreen to allow for selection of multiple entities at once
+ */
+public class MultiSelectEntityScreen extends EntityScreen {
+
+    private int maxSelectValue = -1;
+
+    private final EntitiesSelectionCache entitiesSelectionCache;
+
+    public MultiSelectEntityScreen(boolean handleCaseIndex, boolean full,
+            SessionWrapper session, @Nullable String[] selectedValues,
+            EntitiesSelectionCache entitiesSelectionCache)
+            throws CommCareSessionException, SQLException {
+        super(handleCaseIndex, full, session);
+        this.entitiesSelectionCache = entitiesSelectionCache;
+        processSelectedValues(selectedValues, session);
+    }
+
+    private void processSelectedValues(String[] selectedValues, SessionWrapper session)
+            throws SQLException, CommCareSessionException {
+        if (selectedValues != null) {
```

Comment:
So if all values are to be unselected, this function would get an empty array passed in?

Reply:
I am confused, can we navigate forward on Entity Screen with all values unselected ? I was supposing that can't happen.

Reply:
oh, no, you're right - I was narrowly focused on the code and not thinking about the context